packages/typescript/src/lib/TypescriptCompiler.ts

```diff
@@ -166,7 +166,75 @@ export class TypescriptCompiler {
     }
 
     // eslint-disable-next-line @typescript-eslint/no-var-requires
-    return require(tsconfigPath).compilerOptions;
+    const { compilerOptions } = require(tsconfigPath);
+
+    return this.normalizeTSConfigCompilerOptions(compilerOptions);
+  }
+
+  private normalizeTSConfigCompilerOptions(
+    options: ts.CompilerOptions
+  ): ts.CompilerOptions {
+    if (options.moduleResolution) {
+      const moduleResolution = (
+        options.moduleResolution as unknown as string
+      ).toLowerCase();
+      switch (moduleResolution) {
+        case 'classic':
+          options.moduleResolution = ts.ModuleResolutionKind.Classic;
+          break;
+        case 'node':
+          options.moduleResolution = ts.ModuleResolutionKind.NodeJs;
+          break;
+        case 'node16':
+          options.moduleResolution = ts.ModuleResolutionKind.Node16;
+          break;
+        case 'nodenext':
+          options.moduleResolution = ts.ModuleResolutionKind.NodeNext;
+          break;
+      }
+    }
+
+    if (options.target) {
+      const target = (options.target as unknown as string).toLowerCase();
```

Comment:
According to http://json.schemastore.org/tsconfig, `target` is not a mandatory field.
So, invoking `toLowerCase` could be dangerous.